### ISSUE TITLE
Will there be http2 compatibility with the 2.3.0 client?

### ISSUE THREAD

**Issue by mightyBroccoli** [nextcloud/client_theming#76]
Will there be an option to use http2 with the nextcloud client? 
I think the client and the server would benefit from using http2.
Are there any plans regarding this?

**Comment by Pablohn26** [comment]
I am using nextcloud client with my server serving with http2 and I have not found any problem. I think it is a server configuration rather than client configuration. Have you tested it?  Which is the problem that you have?

**Comment by mightyBroccoli** [comment]
If you look into your apache log file you will see that the client will not connect via http2, the connection is established with http/1.1
For Windows:
`[01/Mar/2017:01:27:19 +0100] "PROPFIND /remote.php HTTP/1.1" 207 1170 "-" "Mozilla/5.0 (Windows) mirall/2.2.4 (build 2) (Nextcloud)"`
For Linux:
`[01/Mar/2017:01:27:11 +0100] "PROPFIND /remote.php HTTP/1.1" 207 1215 "-" "Mozilla/5.0 (Linux) mirall/2.2.4 (Nextcloud)"`

The Server configuration is completely unrelevant for that. The nextcloud is not the "server" just the application which  interprets the answer from the webserver.
I tested it thoroughly and it seems that the QT Library is not able to make a connection with http2. 
[owncloud/client/issues/3146](https://github.com/owncloud/client/issues/3146)
Pls check your log files and i am fairly sure that even your clients will connect through http/1.1

**Comment by benediktg** [comment]
```
[01/Mar/2017:02:01:35 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 802 "-" "Mozilla/5.0 (Linux) mirall/2.2.4 (Nextcloud)" "-"
```
Indeed… It connects via HTTP/1.1 although I have nginx with http2 enabled.

**Comment by Pablohn26** [comment]
@mightyBroccoli from my ignorance, have you ever seen WebDAV over HTTP2?

Its [RFC](https://tools.ietf.org/html/rfc4918) states 

> Web Distributed Authoring and Versioning (WebDAV) consists of a set of methods, headers, and content-types ancillary to HTTP/1.1

so I think there is no WebDAV protocol over HTTP2 as it was just an extension of HTTP/1.1.

**Comment by mightyBroccoli** [comment]
@Pablohn26 The question in this issue is will the be support for the http2 for the client. 
I configured my apache correctly and http2 request are served. Using the webinterface http2 is not problem but the webui is barely used. In most cases the client is used to transfer eg.
That's why I want to know if there will be http2 for the client in any way shape or form.

**Comment by Bullnados** [comment]
Is there a benefit for http2 in client?

**Comment by tessus** [comment]
What happens if someone only activates http2 on the server? Will the client stop working?

**Comment by mightyBroccoli** [comment]
@Bullnados http2 is the complete overhaul of http/1.1 there is not just one special feature that is better than http1.1 it is the new version of http. Explaining the difference here would be humongous so look here [http1 vs http2](http://stackoverflow.com/questions/28592077/difference-between-http-1-1-and-http-2-0)

@tessus I never tried that. But the nextcloud client is based on the qt networking and qt networking is not yet able to use http2 thats why the connection will not work with just http2 enabled, at least that is my guess.

**Comment by Bullnados** [comment]
@mightyBroccoli 
Thanks for the link, I found als that one https://github.com/bagder/http2-explained. It is availible in multiple languages.

But the question is not if http2 will be implemented in feature, of course it should be, but if http2 have benefits for the client. F.e. has multiplexing a benefit for clinet sync?

**Comment by enoch85** [comment]
So what's the result here. Is it supported?

**Comment by Bullnados** [comment]
Nohttps://github.com/nextcloud/client_theming/issues/76#issuecomment-283206595 and ff

**Comment by mightyBroccoli** [comment]
@enoch85 the result is not yet. The http2 compatibility will be released with with qt 5.8 

As long the build process will use 5.8 someday http2 will come. Depends in how fast the implementation of 5.8 takes.

**Comment by C0rn3j** [comment]
Qt 5.8 is already out, seems like it was released 3 months ago?

https://wiki.qt.io/Qt_5.8_Release

**Comment by mightyBroccoli** [comment]
It seems that they managed to implement http2 in Qt 5.8. 

[Changelog](https://wiki.qt.io/New_Features_in_Qt_5.8)
```
Qt Network
- Support for TLS PSK ciphersuites on server side
- Support for configurable diffie hellman parameters
- Support for HTTP/2 in Network Access Manager
```

Does this mean the client will receive http2 sometime in the future?

**Comment by enoch85** [comment]
As the Nextcloud Desktop client "only" is a themed  ownCloud client, I think this question is best to ask in the ownCloud client repo.

**Comment by enoch85** [comment]
This issue has been inactive for some time now. 

Is this issue related to the ownCloud client, (thus not an **theming** isssue) please close it and report the issue [here](https://github.com/owncloud/client) instead.
Is this issue related to this repo but solved, please close it and if possible let us know what solved the issue.
Is this issue still not solved, please let us know as well.

If I don't hear anything from the one who created this issue within 2 weeks, I will close the issue.

**Comment by mightyBroccoli** [comment]
I think this issue is resolved with owncloud/client/pull/5659.
